### PR TITLE
Bugfix: line invalidation for update with no operations

### PR DESCRIPTION
**Pull request by scholtzan**
This is required to support changes made in https://github.com/xi-editor/xi-editor/pull/1236

Without this fix there are line invalidation issues.

### CODE REVIEW

**Comment by scholtzan** [comment]
```error: Could not compile `bytecount`.``` Why? :(

**Comment by nangtrongvuon** [comment]
Maybe trigger another rebuild? This seems like a rust thing from what little I've looked at online.

**Comment by cmyr** [comment]
@scholtzan try updating the CI script to run on stable rust, instead of being pinned to 1.31?

File: Tests/IntegrationTests/SearchIntegrationTests.swift [hunk 1]
```diff
@@ -17,74 +17,79 @@ import XCTest
...
-
+    
     override func tearDown() {
         super.tearDown()
         searchTester = nil
     }
 
     func testSearchingGoldenBoys() {
-        let findExpectation = expectation(description: "find expectation")
+        weak var findExpectation = expectation(description: "find expectation")
```
Comment (scholtzan) [inline Tests/IntegrationTests/SearchIntegrationTests.swift]: I got the following error before: `API violation - multiple calls made to -[XCTestExpectation fulfill] for find expectation.`

**Comment by scholtzan** [comment]
Yep, changing it to `stable` fixed it

File: Tests/IntegrationTests/SearchIntegrationTests.swift [hunk 2]
```diff
@@ -17,74 +17,79 @@ import XCTest
 
 
 class SearchIntegrationTests: XCTestCase {
-
```
Comment (nangtrongvuon) [inline Tests/IntegrationTests/SearchIntegrationTests.swift]: nit: whitespace

File: Tests/IntegrationTests/SearchIntegrationTests.swift [hunk 3]
```diff
@@ -17,74 +17,79 @@ import XCTest
...
+    
     var searchTester: SearchTester?
-
```
Comment (nangtrongvuon) [inline Tests/IntegrationTests/SearchIntegrationTests.swift]: nit: whitespace

**Pull request merged**